### PR TITLE
DIUX imagenet/cifar runs

### PR DESCRIPTION
**Pull request by yaroslavvb**
Running TensorPack baselines on g3/p3 instances

### CODE REVIEW

**Comment by codyaustun** [comment]
@yaroslavvb it looks like the hours and epoch columns are mislabelled. Are they supposed to be flipped?

**Comment by yaroslavvb** [comment]
fixed

File: CIFAR10/train/diux_cifar_g3.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    "version": "v1.0",
+    "author": "DIUX",
+    "authorEmail": "ybulatov.ctr@diux.mil",
+    "framework": "tensorflow 1.5, tensorpack 0.8.1",
+    "codeURL": "https://github.com/ppwwyyxx/tensorpack/blob/master/examples/ResNet/cifar10-resnet.py",
+    "model": "ResNet50",
+    "hardware": "g3.4xlarge",
+    "timestamp": "2018-1-31",
+    "misc": "ami-9ba7c4e1 (Amazon Deep Learning AMI version 2, Ubuntu, conda based) "
```
Comment (codyaustun) [inline CIFAR10/train/diux_cifar_g3.json]: You're missing a comma at the end of this line. Also, "misc" should be an object. Please include a key for this value. Here is an [example](https://github.com/stanford-futuredata/dawn-bench-entries/blob/844790a9e4cae17eb545b8fd67b9150a55219a1c/CIFAR10/train/dawn_resnet164_b_1k80-gc_pytorch.json#L11). Also, can you include your learning rate schedule and optimizer. From your, codeURL it looks like it should be:
```
"optimizer": "SGD with Momentum",
"momentum": 0.9,
"schedule": [
    {"learning_rate": 0.1, "epochs": 82},
    {"learning_rate": 0.01, "epochs": 41},
    {"learning_rate": 0.001, "epochs": 177},
    {"learning_rate": 0.0002, "epochs": 100}
],
# Please check that this is valid JSON before committing
```

File: CIFAR10/train/diux_cifar_p3.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+    "version": "v1.0",
+    "author": "DIUX",
+    "authorEmail": "ybulatov.ctr@diux.mil",
+    "framework": "tensorflow 1.5, tensorpack 0.8.1",
+    "codeURL": "https://github.com/ppwwyyxx/tensorpack/blob/master/examples/ResNet/cifar10-resnet.py",
+    "model": "ResNet50",
+    "hardware": "p3.2xlarge",
+    "timestamp": "2018-1-31",
+    "misc": "ami-9ba7c4e1 (Amazon Deep Learning AMI version 2, Ubuntu, conda based) "
```
Comment (codyaustun) [inline CIFAR10/train/diux_cifar_p3.json]: Same comment as above

File: ImageNet/train/diux_resnet152_p3-tensorpack.csv [hunk 3]
```diff
@@ -0,0 +1,12 @@
+{
+    "version": "v1.0",
+    "author": "DIUX",
+    "authorEmail": "ybulatov.ctr@diux.mil",
+    "framework": "tensorflow 1.5, tensorpack 0.8.1",
+    "codeURL": "https://github.com/diux-dev/cluster/blob/master/dawn/resnet.b512.baseline.py",
+    "model": "ResNet152",
+    "hardware": "p3.16xlarge",
+    "timestamp": "2018-1-31",
+    "misc": "ami-9ba7c4e1 (Amazon Deep Learning AMI version 2, Ubuntu, conda based) "
```
Comment (codyaustun) [inline ImageNet/train/diux_resnet152_p3-tensorpack.csv]: Same as above, but different learning schedule: 
```
"optimizer": "SGD with Momentum",
"momentum": 0.9,
"schedule": [
    {"learning_rate": 0.1, "epochs": 1},
    {"learning_rate": 0.15, "epochs": 1},
    {"learning_rate": 0.2, "epochs": 28},
    {"learning_rate": 0.02, "epochs": 30},
    {"learning_rate": 0.002, "epochs": 20},
    {"learning_rate": 0.0002, "epochs": 20}
]
# Please check that this is valid JSON before committing
```

This might be wrong. Please double check. I am getting the base learning rate from [L34](https://github.com/diux-dev/cluster/blob/2c444ab014f21ee97642bdec043ef0c6ed884541/dawn/resnet.b512.baseline.py#L34)

**Review by codyaustun — CHANGES_REQUESTED** [review]

File: CIFAR10/train/diux_cifar_g3.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+    "version": "v1.0",
+    "author": "DIUX",
+    "authorEmail": "ybulatov.ctr@diux.mil",
+    "framework": "tensorflow 1.5, tensorpack 0.8.1",
+    "codeURL": "https://github.com/ppwwyyxx/tensorpack/blob/master/examples/ResNet/cifar10-resnet.py",
+    "model": "ResNet50",
+    "hardware": "g3.4xlarge",
+    "timestamp": "2018-1-31",
```
Comment (codyaustun) [inline CIFAR10/train/diux_cifar_g3.json]: Can you update the dates to %Y-%m-%d format, i.e. 2018-01-31?

**Comment by yaroslavvb** [comment]
added formatting/schedule and ran through https://jsonlint.com/, ptal

File: CIFAR10/train/diux_cifar_g3.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    "authorEmail": "ybulatov.ctr@diux.mil",
+    "framework": "tensorflow 1.5, tensorpack 0.8.1",
+    "codeURL": "https://github.com/ppwwyyxx/tensorpack/blob/master/examples/ResNet/cifar10-resnet.py",
+    "model": "ResNet50",
+    "hardware": "g3.4xlarge",
+    "timestamp": "2018-01-31",
+    "costPerHour": 1.14,
+    "optimizer": "SGD with Momentum",
+    "momentum": 0.9,
+    "schedule": [
```
Comment (codyaustun) [inline CIFAR10/train/diux_cifar_g3.json]: Can you move schedule into the misc object:

```
{
	"version": "v1.0",
	"author": "DIUX",
	"authorEmail": "ybulatov.ctr@diux.mil",
	"framework": "tensorflow 1.5, tensorpack 0.8.1",
	"codeURL": "https://github.com/ppwwyyxx/tensorpack/blob/master/examples/ResNet/cifar10-resnet.py",
	"model": "ResNet50",
	"hardware": "g3.4xlarge",
	"timestamp": "2018-01-31",
	"costPerHour": 1.14,
	"optimizer": "SGD with Momentum",
	"momentum": 0.9,
	"misc": {
		"AMI": "ami-9ba7c4e1",
		"schedule": [{
				"learning_rate": 0.1,
				"epochs": 82
			},
			{
				"learning_rate": 0.01,
				"epochs": 41
			},
			{
				"learning_rate": 0.001,
				"epochs": 177
			},
			{
				"learning_rate": 0.0002,
				"epochs": 100
			}
		]
	}
}
```

Note this includes the correction to my comment below

File: CIFAR10/train/diux_cifar_g3.json [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+        {"learning_rate": 0.1, "epochs": 1},
+        {"learning_rate": 0.01, "epochs": 82},
+        {"learning_rate": 0.001, "epochs": 123},
+        {"learning_rate": 0.0002, "epochs": 300}
```
Comment (codyaustun) [inline CIFAR10/train/diux_cifar_g3.json]: The format of the CIFAR10 and ImageNet schedules are inconsistent. Here epochs seem to mean the epoch where the change in learning rate occurs. More concretely, are you saying the learning schedule is the following?

1. Learning rate = 0.1 from epoch 1 to epoch 82
2. Learning rate = 0.01 from epoch 83 to 123
3. Learning rate = 0.001 from epoch 124 to 300
4. Learning rate = 0.0002 from epoch 300 to 400, which is the end of the TSV

The ImageNet JSON uses epochs as the number of epochs for the specified learning rate, so adding up the epochs fields in schedule adds up to the total number of epochs in the TSV. I prefer the approach from the ImageNet JSON. I think this should be:

```
        "schedule": [
                {"learning_rate": 0.1, "epochs": 82},
                {"learning_rate": 0.01, "epochs": 41},
                {"learning_rate": 0.001, "epochs":  177},
                {"learning_rate": 0.0002, "epochs": 100}
            ]
```

82 + 41 + 177 + 100 == 400 matching the TSV

File: ImageNet/train/diux_resnet152_p3-tensorpack.json [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+  "schedule": [
+    {"learning_rate": 0.1, "epochs": 1},
+    {"learning_rate": 0.15, "epochs": 1},
+    {"learning_rate": 0.2, "epochs": 28},
+    {"learning_rate": 0.02, "epochs": 30},
+    {"learning_rate": 0.002, "epochs": 20},
+    {"learning_rate": 0.0002, "epochs": 20}
+  ],
+	"misc": {
+		"ami": "ami-9ba7c4e1 (Amazon Deep Learning AMI version 2, Ubuntu, conda based)"
```
Comment (codyaustun) [inline ImageNet/train/diux_resnet152_p3-tensorpack.json]: You used "AMI" for the CIFAR10 experiments. It would be nice if they were consistent.

File: ImageNet/train/diux_resnet152_p3-tensorpack.json [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+	"authorEmail": "ybulatov.ctr@diux.mil",
+	"framework": "tensorflow 1.5, tensorpack 0.8.1",
+	"codeURL": "https://github.com/diux-dev/cluster/blob/master/dawn/resnet.b512.baseline.py",
+	"model": "ResNet152",
+	"hardware": "p3.16xlarge",
+	"timestamp": "2018-01-31",
+  "costPerHour": 24.48,
+  "optimizer": "SGD with Momentum",
+  "momentum": 0.9,
+  "schedule": [
```
Comment (codyaustun) [inline ImageNet/train/diux_resnet152_p3-tensorpack.json]: Like the comment above I would move the schedule field into misc:

```
{
	"version": "v1.0",
	"author": "DIUX",
	"authorEmail": "ybulatov.ctr@diux.mil",
	"framework": "tensorflow 1.5, tensorpack 0.8.1",
	"codeURL": "https://github.com/diux-dev/cluster/blob/master/dawn/resnet.b512.baseline.py",
	"model": "ResNet152",
	"hardware": "p3.16xlarge",
	"timestamp": "2018-01-31",
	"costPerHour": 24.48,
	"optimizer": "SGD with Momentum",
	"momentum": 0.9,
	"misc": {
		"AMI": "ami-9ba7c4e1 (Amazon Deep Learning AMI version 2, Ubuntu, conda based)",
		"schedule": [{
				"learning_rate": 0.1,
				"epochs": 1
			},
			{
				"learning_rate": 0.15,
				"epochs": 1
			},
			{
				"learning_rate": 0.2,
				"epochs": 28
			},
			{
				"learning_rate": 0.02,
				"epochs": 30
			},
			{
				"learning_rate": 0.002,
				"epochs": 20
			},
			{
				"learning_rate": 0.0002,
				"epochs": 20
			}
		]
	}
}
```

File: CIFAR10/train/diux_cifar_p3.json [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+    "authorEmail": "ybulatov.ctr@diux.mil",
+    "framework": "tensorflow 1.5, tensorpack 0.8.1",
+    "codeURL": "https://github.com/ppwwyyxx/tensorpack/blob/master/examples/ResNet/cifar10-resnet.py",
+    "model": "ResNet50",
+    "hardware": "p3.2xlarge",
+    "timestamp": "2018-01-31",
+    "costPerHour": 3.06,
+    "optimizer": "SGD with Momentum",
+    "momentum": 0.9,
+    "schedule": [
```
Comment (codyaustun) [inline CIFAR10/train/diux_cifar_p3.json]: Same comments as the other CIFAR10 experiments. I think this should be:

```
{
	"version": "v1.0",
	"author": "DIUX",
	"authorEmail": "ybulatov.ctr@diux.mil",
	"framework": "tensorflow 1.5, tensorpack 0.8.1",
	"codeURL": "https://github.com/ppwwyyxx/tensorpack/blob/master/examples/ResNet/cifar10-resnet.py",
	"model": "ResNet50",
	"hardware": "p3.2xlarge",
	"timestamp": "2018-01-31",
	"costPerHour": 3.06,
	"optimizer": "SGD with Momentum",
	"momentum": 0.9,
	"misc": {
		"AMI": "ami-9ba7c4e1",
		"schedule": [{
				"learning_rate": 0.1,
				"epochs": 82
			},
			{
				"learning_rate": 0.01,
				"epochs": 41
			},
			{
				"learning_rate": 0.001,
				"epochs": 177
			},
			{
				"learning_rate": 0.0002,
				"epochs": 100
			}
		]
	}
}
```

**Review by codyaustun — CHANGES_REQUESTED** [review]

**Comment by codyaustun** [comment]
Aside from the comments in my last review, everything looks good.

**Comment by yaroslavvb** [comment]
incorporated changes and ran through json validator (btw, you could probably make some of these sanity checks run automatically for each PR with Travis)

**Review by codyaustun — APPROVED** [review]

**Comment by codyaustun** [comment]
Great suggestion. I think it is worth adding.
